### ISSUE TITLE
Toll Free Numbers are Excluded when Generating Data

### ISSUE THREAD

**Issue by ooglek** [cedvdb/dart_phone_metadata#8]
Your parsed metadata does not include the following types of numbers:

- tollFree
- premiumRate
- sharedCost
- uan
- pager
- personalNumber
- voip
- noInternationalDialing
- voicemail

The problem with this is that [cedvdb/phone_numbers_parser](https://github.com/cedvdb/phone_numbers_parser) `isValid()` now returns `false` for `+1 800 432 1000` which is not true, that is a valid US Toll Free number.

**Comment by cedvdb** [comment]
Yes, this library only supports mobile and fixed line (currently). The other types were way out of scope for my personal use cases at the time.

Do you have an use case ? Obviously there are valid use cases, but it's easier to understand with specifics.

**Comment by ooglek** [comment]
Yep -- I run a phone company, and I need a way in-app using Flutter to ensure that the phone number(s) entered in the app are valid (to the best ability of the available LibPhoneNumber / PhoneKit data) before allowing a call or SMS to continue.

Another app implemented LibPhoneNumber on Android and PhoneKit on IOS through direct access to those systems, but in our testing, the same code on Android produced different results than IOS, which made testing and debugging a giant pain in the butt, having to understand the finer points of formatting and testing validity between the two platforms.

Your code simplified this and was not asynchronous, and was very fast. Thus the hope to use it. 

But if the underlying data will only ever support mobile and fixedLine, then it cannot be used, sadly.

I'm happy to help maintain it and keep it up to date, but if your interest is not there to do so, rather than build and maintain my own, I'll likely find another project.

**Comment by cedvdb** [comment]
> I'm happy to help maintain it and keep it up to date, but if your interest is not there to do so, rather than build and maintain my own, I'll likely find another project.

My interest is there, else I wouldn't have done all this so far 😉. So don't worry about that and I personally would like to keep this repository not split between different implementations as it would mean less open source additions like yours and other people. This is not the first phone company that seems to have an use case for this library too so obviously this is an important feature. So I'm encouraging you and others to keep the efforts constrained here for classic open source reasons.

Now concerning the tree shaking, I think this is important to take into account for other people that might have a lighter use than that. I'll reiterate that this might be a non issue, but should be cleared in case it is one.

**Comment by ooglek** [comment]
I'm not sure adding some code to the dart file is going to kill memory usage. We're talking about 254 or so countries, and each are going to have a different combo of number types. I added it up:

- 254 generalDesc
- 247 mobile
- 245 fixed line
- 184 tollFree
- 143 premiumRate
- 94 voip
- 65 uan
- 59 personalNumber
- 54 sharedCost
- 33 noInternationalDialling
- 23 pager
- 16 voicemail

Mobile + Fixed Line = 492 regular expressions
All the rest = 671 regular expressions

So you're adding 136% more regular expressions. But it's just an object, and frankly that shouldn't really impact memory usage much at all. 

What audience wants a lighter use? And how much lighter is the code incomplete than it is complete? E.g. `isValid()` returns a lie unless you KNOW the phone number is mobile or fixedLine. If your goal is just to check if it is mobile, sure, but in the US and other countries, Number Portability changes things entirely. You cannot guarantee that a phone number is or is not mobile vs fixedLine because you can turn any US (or any +1 country) number from a fixedLine (wireline, landline) to a mobile and vice versa.

I'm happy to keep it constrained to actual phone numbers, and the data on which this library is built -- LibPhoneNumber / PhoneKit. I'm not sure what you mean by "keep the efforts constrained..." help? My goal is a library that works the same in Android as it does on IOS (and the web) using regularly updated Metadata from LibPhoneNumber and/or PhoneKit. But it has to implement ALL the formats and ALL the validity checks.

Until it covers 100% of the LibPhoneNumber data, I think the project should make it clear that the data is only partially implemented.

If your vision and commitment is to implement 100% of the data, I'm happy to contribute! If it might take a few weeks to happen and/or do approvals or PRs, then I may have to move on. My timeline issue, not yours! And I am in no way expecting nor demanding your time to do so, only asking if it is a reasonable timeline for you.

**Comment by cedvdb** [comment]
@ooglek unfortunately, that will not be a reasonable time line for me (not even close) as I have priorities that far outweigh this project currently.

I agree that the library should make it clearer that it only supports a subset currently.

 > and how much lighter is the code incomplete than it is complete? 

<< it might be a non issue >>. The point is we don't know. Adding a few hundred KB would be quite unreasonable for a simple phone number input that only accepts mobile phone numbers, especially for the web, which is most of the use case I imagined this library would be used for when I designed this library originally as I thought if more processing was required it would have probably been done on the backend.

**Comment by ooglek** [comment]
Right on! I'm going to fork the repos and see if I can get it working. Happy to contribute back to your repo as well, obviously!

**Comment by cedvdb** [comment]
Yes, I'm available for reviews, just not for more time consuming tasks. Thank you.
